### PR TITLE
Rewrite library: revamp UI (3/4)

### PR DESCRIPTION
**Pull request by SUPERCILEX**
I got everything done on my side including adding the changelog and updating the README—I'm just submitting these PRs so they're ready.

### CODE REVIEW

**Comment by yshrsmz** [comment]
Could you change the target branch & rebase?

File: library/src/main/res/values/ids.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <item name="colorPrimary" type="color" />
+    <item name="colorPrimaryDark" type="color" />
+    <item name="colorAccent" type="color" />
```
Comment (SUPERCILEX) [inline library/src/main/res/values/ids.xml]: These colors are now required. We could add defaults if you'd like.

File: library/src/main/java/net/yslibrary/licenseadapter/LicenseAdapter.java [hunk 2]
```diff
@@ -55,110 +58,34 @@ public LicenseViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
 
-  @SuppressWarnings("unchecked")
   @Override
-  public void onBindViewHolder(final LicenseViewHolder holder, final int position) {
-    final Wrapper wrapper = getItem(position);
-
-    if (wrapper == null) {
-      throw new IllegalStateException("No wrapper for this position: " + position);
+  public void onBindViewHolder(ViewHolderBase holder, int position) {
+    if (position % 2 == 0) {
```
Comment (yshrsmz) [inline library/src/main/java/net/yslibrary/licenseadapter/LicenseAdapter.java]: we should use `getItemViewType` instead

File: library/src/main/res/layout/license.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+             xmlns:tools="http://schemas.android.com/tools"
+             android:layout_width="match_parent"
+             android:layout_height="wrap_content"
+  >
+
+  <!--
+  Wrap this LL in a FL to circumvent gone bug where a top-level LL will still be used in onMeasure
+  event though it's marked as gone.
```
Comment (yshrsmz) [inline library/src/main/res/layout/license.xml]: Is there any link to the issue, like stackoverflow or google issue tracker?
including those links would be helpful.

File: library/src/main/res/values/ids.xml (hunk 1 above)
Comment (yshrsmz) [reply library/src/main/res/values/ids.xml]: I think defaults are not needed if we make sure to note that in README.

Comment (SUPERCILEX) [reply library/src/main/res/values/ids.xml]: Sweet!

File: library/src/main/java/net/yslibrary/licenseadapter/LicenseAdapter.java (hunk 2 above)
Comment (SUPERCILEX) [reply library/src/main/java/net/yslibrary/licenseadapter/LicenseAdapter.java]: Oops, yep.

File: library/src/main/res/layout/license.xml (hunk 3 above)
Comment (SUPERCILEX) [reply library/src/main/res/layout/license.xml]: Done.

**Comment by yshrsmz** [comment]
Thanks! 👍

**Comment by SUPERCILEX** [comment]
Awesome sauce, there's only #35 left to go! 😄

**Comment by turfbuccaneer** [comment]
I don't know what any of that means I wish I did but I don't and that's the
truth sorry

rendosmoke

On Dec 6, 2017 4:03 PM, "Alex Saveau" <notifications@github.com> wrote:

Awesome sauce, there's only #35
<https://github.com/yshrsmz/LicenseAdapter/pull/35> left to go! 😄

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/yshrsmz/LicenseAdapter/pull/34#issuecomment-349816342>,
or mute the thread

**Pull request merged**